test/e2e/k8s_util.go

```diff
@@ -700,20 +701,21 @@ func (k *KubernetesUtils) waitForPodInNamespace(ns string, pod string) ([]string
 func (k *KubernetesUtils) waitForHTTPServers(allPods []Pod) error {
 	const maxTries = 10
 	log.Infof("waiting for HTTP servers (ports 80, 81 and 8080:8085) to become ready")
-	log.Infof("Waiting for HTTP servers (ports 80, 81, 5000 and 8080:8085) to become ready")
 	var wrong int
 	for i := 0; i < maxTries; i++ {
 		reachability := NewReachability(allPods, Connected)
-		k.Validate(allPods, reachability, 80, v1.ProtocolTCP)
-		k.Validate(allPods, reachability, 80, v1.ProtocolUDP)
-		k.Validate(allPods, reachability, 80, v1.ProtocolSCTP)
-		k.Validate(allPods, reachability, 81, v1.ProtocolTCP)
-		k.Validate(allPods, reachability, 81, v1.ProtocolUDP)
-		k.Validate(allPods, reachability, 81, v1.ProtocolSCTP)
-		for j := 8080; j < 8086; j++ {
-			k.Validate(allPods, reachability, int32(j), v1.ProtocolTCP)
-		}
-		_, wrong, _ = reachability.Summary()
+		curWrong := 0
+
+		k.Validate(allPods, reachability, []int32{80, 81, 8080, 8081, 8082, 8083, 8084, 8085}, v1.ProtocolTCP)
+		_, curWrong, _ = reachability.Summary()
```

Comment:
Shouldn't the logic here be: for any reachability matrix that has wrong results, i++ and sleep immediately, instead of continue doing unnecessary probes?

Reply:
Done.